src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -13,8 +13,12 @@
     /// </summary>
     internal class HostingDiagnosticListener : IApplicationInsightDiagnosticListener
     {
+        /// <summary>
+        /// Conversion from stopwatch time (ticks since boot) to local time.
+        /// </summary>
+        private static readonly DateTimeOffset stopwatchEpoch = DateTime.Now - TimeSpan.FromTicks(Stopwatch.GetTimestamp());
```

Comment:
We found issues with this approach - it will work badly when windows adjusts the clock (some machines has bad bios and time constantly running behind or ahead) or when there is a summer time switch. Mostly the first though.

You can use extension methods `RequestTelemetry.Start()` and `Stop()` instead. See https://github.com/Microsoft/ApplicationInsights-dotnet/blob/dd654d2c0cd47623d0e99e6e19481cba3203fa98/src/Core/Managed/Shared/OperationTelemetryExtensions.cs#L19